indico/modules/rb/controllers/user/rooms.py

```diff
@@ -156,19 +158,45 @@ def _checkParams(self):
         self._room = Room.get(request.view_args['roomID'])
         self._occupancy_period = request.args.get('period', 'pastmonth')
         self._end = date.today()
+        year = None
+        month = None
         if self._occupancy_period == 'pastmonth':
             self._end = self._end - relativedelta(days=1)
             self._start = self._end - relativedelta(days=29)
         elif self._occupancy_period == 'thisyear':
             self._start = date(self._end.year, 1, 1)
         elif self._occupancy_period == 'sinceever':
-            self._start = Reservation.query.with_entities(func.min(Reservation.start_dt)).one()[0].date()
+            oldest = db.session.query(func.min(Reservation.start_dt)).filter_by(room_id=self._room.id).one()[0]
+            self._start = oldest_booking_date = oldest.date() if oldest else self._end
         else:
-            raise IndicoError('Invalid period specified')
+            match = re.match(r'(\d{4})(?:-(\d{2}))?', self._occupancy_period)
+            if match is None:
+                raise IndicoError('Invalid period specified')
+            year = int(match.group(1))
+            month = int(match.group(2)) if match.group(2) else None
+            if month:
+                try:
+                    self._start = date(year, month, 1)
+                except ValueError:
+                    raise IndicoError('Invalid year or month specified')
+                self._end = self._start + relativedelta(months=1)
+                self._occupancy_period = '{:d}-{:02d}'.format(year, month)
+            else:
+                try:
+                    self._start = date(year, 1, 1)
+                except:
```

Comment:
never use a bare `except:` - this catches exceptions such as SystemExit and KeyboardInterrupt. `except Exception:` would avoid this but I'm pretty sure it's enough to catch `ValueError` here.